thorlcr/master/thgraphmanager.cpp

```diff
@@ -838,8 +838,21 @@ void CJobManager::reply(IConstWorkUnit *workunit, const char *wuid, IException *
     else if (e)
     {
         IThorException *te = QUERYINTERFACE(e, IThorException);
-        if (te && TE_WorkUnitAborting == te->errorCode())
-            replyMb.append((unsigned)DAMP_THOR_REPLY_ABORT);
+        if (te)
+        {
+            switch (te->errorCode())
+            {
+            case TE_WorkUnitAborting:
+                replyMb.append((unsigned)DAMP_THOR_REPLY_ABORT);
+                break;
+            case TE_CostExceeded:
+                replyMb.append((unsigned)DAMP_THOR_REPLY_COSTEXCEEDED);
+                break;
```

Comment:
I don't think you need a new reply code.

A user abort is thrown as:
```
MakeThorException(TE_AbortException, "THOR ABORT")
```

vs your exception is:
```
                LOG(MCwarning, thorJob, "ABORT job cost exceeds limit");
                Owned<IException> e = MakeThorException(TE_CostExceeded, "Job cost exceeds limit (cost: %" I64F "d, limit %u)", totalCost, costLimit);
```

But all failure exceptions from Thor will take this path, resulting in them being reported to the workunit with their text.
TE_WorkUnitAborting is special case - I'm not sure why it was introduced.